src/lib/viewers/image/ImageViewer.js

```diff
@@ -507,6 +508,22 @@ class ImageViewer extends ImageBaseViewer {
         this.annotator.toggleAnnotationMode(nextMode);
         this.processAnnotationModeChange(nextMode);
     }
+
+    handleAnnotationCreateEvent({ annotation: { id } = {}, meta: { status } = {} }) {
+        // Only on success do we exit create annotation mode. If error occurs,
+        // we remain in create mode
+        if (status === 'success') {
+            this.annotator.emit('annotations_active_set', id);
+        }
+    }
+
+    initAnnotations() {
+        super.initAnnotations();
+
+        if (this.areNewAnnotationsEnabled() && this.annotationControls) {
```

Comment:
Should we be checking `this.annotator`, instead or as well?

Reply:
`BaseViewer.initAnnotations` already assumes `this.annotator` exists, so it seems safe enough?

Reply:
I'm confused why the `this.annotationControls` check is needed, then. It's not referenced in the subsequent code or in the handler.

Reply:
Will remove that check since it seems unnecessary